### ISSUE TITLE
Dockerfile rules - false positive on multi-stage build with CMDs

### ISSUE THREAD

**Issue by ewiner** [returntocorp/sgrep-rules#1520]
`generic.dockerfile.correctness.multiple-cmd-instructions.multiple-cmd-instructions`

**Describe the bug**
A clear and concise description of what the bug is and for which rule. Is the rule producing a false positive? Is it incorrect or missing something?

https://semgrep.dev/r?q=generic.dockerfile.correctness.multiple-cmd-instructions.multiple-cmd-instructions is flagging multiple CMD instructions, even when those CMDs are each in a different stage of the build.

**To Reproduce**

```Dockerfile
FROM busybox AS stage1
CMD /bin/true
FROM stage1 AS stage2
CMD /bin/false
```

**Expected behavior**

This rule should only fire if there's two `CMD`s within a stage (between `FROM` instructions).

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [x] P2: annoying but not blocking me

**Comment by minusworld** [comment]
@ewiner Good catch! Thanks! Does https://github.com/returntocorp/semgrep-rules/pull/1521 look like it will address the issue? :D

**Comment by ewiner** [comment]
Yep!

**Comment by minusworld** [comment]
@ewiner Awesome, thanks! Let me know if you encounter any more issues :D
